apis/kueue/v1beta1/workload_types.go

```diff
@@ -336,6 +359,11 @@ const (
 // +kubebuilder:resource:shortName={wl}
 
 // Workload is the Schema for the workloads API
+// +kubebuilder:validation:XValidation:rule="has(self.spec) && has(self.status) && has(self.status.conditions) && self.status.conditions.exists(c, c.type == 'QuotaReserved' && c.status == 'True') && has(self.status.admission) ? size(self.spec.podSets) == size(self.status.admission.podSetAssignments) : true", message="podSetAssignments must have the same number of podSets as the spec"
```

Comment:
Dropping `has(self.status)`  check causes most of the workload tests to fail due to unexisting `status` field. For example [this](https://github.com/kubernetes-sigs/kueue/blob/8be7044e52268654d7daca5dc5b3e3d8eac0f007/test/integration/webhook/workload_test.go#L157) is one of the test cases which fails when skipping the status check.

An alternative could be using CEL optional types, that way we can leave just `has(oldSelf.?status.conditions)`. However, optional types require Kubernetes 1.29. 

Should I leave it as is and add this issue to #1986?

Reply:
I've removed `has(self.spec)` successfully, however, it seems we still need `has (self.status.conditions)` since `self.status.conditions.exists()` returns the error `no such key: conditions` when conditions are empty.

Reply:
Ah gotcha.

I still see `has(oldSelf.spec)`, however.

Reply:
The updated rules without the spec check should be visible now :)